_posts/2017-03-28-table-sql-api-update.md

```diff
@@ -0,0 +1,195 @@
+date:   2017-03-27 12:00:00
+author: "Timo Walther"
+author-twitter: "twalthr"
+categories: news
+---
+Here’s the thing about stream processing: it offers a lot of value, but it sure can be hard. Many organizations have recognized the benefit of managing large volumes of data in real-time, reacting quickly to trends, and providing customers with live services at scale. Streaming applications with well-defined business logic can deliver a competitive advantage.
+
+But streaming concepts such as windows, event time, out-of-order event handling, and state management are challenging for many users who could benefit from streaming. Most streaming APIs require deep Java and Scala knowledge, especially when it comes to efficient programming, thus limiting the range of users who can interact with modern streaming frameworks.
+
+To deliver the power of stream processing to a broader set of users, the Apache Flink community is developing APIs that provide simpler abstractions and more concise syntax so that users can focus on their business logic instead of advanced streaming concepts. Along with other APIs (such as [CEP](https://ci.apache.org/projects/flink/flink-docs-release-1.2/dev/libs/cep.html) for complex event processing on streams), Flink offers a relational API that aims to unify stream and batch processing: the [Table & SQL API](https://ci.apache.org/projects/flink/flink-docs-release-1.2/dev/table_api.html), often referred to as the Table API.
+
+Recently, contributors working for companies such as Alibaba, Huawei, data Artisans, and more decided to further develop the Table API. Over the past year, the Table API has been rewritten entirely. Since Flink 1.1, its core has been based on [Apache Calcite](http://calcite.apache.org/), which parses SQL and optimizes all relational queries. Today, the Table API can address a wide range of use cases in both batch and stream environments with unified semantics.
+
+This blog post summarizes the current status of Flink’s Table API and showcases some of the recently-added features in Flink 1.2.0. Among the features presented here are the unified access to batch and streaming data, data transformation, and window operators.
+The following paragraphs are not only supposed to give you a general overview of the Table API, but also to illustrate the potential of relational APIs in the future.
+
+Because the Table API is built on top of Flink’s core APIs, [DataStream](https://ci.apache.org/projects/flink/flink-docs-release-1.2/dev/datastream_api.html) or [DataSet](https://ci.apache.org/projects/flink/flink-docs-release-1.2/dev/batch/index.html) programs can be converted to a Table and vice-versa without much overhead. Hereafter, we show how to create tables from different sources and specify programs that can be executed locally or in a distributed setting. In this post, we will use the Scala version of the Table API, but there is also a Java version as well as a SQL API with an equivalent set of features.
```

Comment:
DataStream and DataSet programs can be converted into a Table -> DataStreams and DataSets can be converted into a Table